dlt/common/storages/fsspec_filesystem.py

```diff
@@ -185,7 +185,12 @@ def open(self, mode: str = "rb", **kwargs: Any) -> IO[Any]:  # noqa: A003
                 **text_kwargs,
             )
         else:
-            opened_file = self.fsspec.open(self["file_url"], mode=mode, **kwargs)
+            opened_file = self.fsspec.open(
```

Comment:
yeah this is close to what we want. but please allow people to switch this behavior on or off:
- add an argument `compression` with `auto` (default - behavior exactly as below), "enable", "disable" to hardcore enable / diable